Integration with backend

josejulio opened the pull request:
![Peek 2020-09-04 08-55](https://user-images.githubusercontent.com/3845764/92247395-ccfa2880-ee8c-11ea-970c-509cc4c6ebb0.gif)

adamruzicka commented on src/PresentationalComponents/NotificationActions/NotificationActions.js:
```diff
@@ -8,9 +8,9 @@ import {
 } from '@patternfly/react-core';
 
-class NotificationActions extends React.Component {
+export class NotificationActions extends React.Component {
```
I assume all these `export`s are to be able to use the not-yet-connected-to-store component in the tests, correct?

bastilian replied:
Correct. :)

bastilian commented on config/setupTests.js:
```diff
@@ -1,10 +1,13 @@
 global.render = render;
 global.mount = mount;
 global.React = React;
+global.insights = {};
```
This was overlooked, it isn't needed, but seemed to be at somepoint when getting the tests to run...

bastilian commented on src/PresentationalComponents/NotificationActions/NotificationActions.js:
```diff
@@ -8,9 +8,9 @@ import {
+export class NotificationActions extends React.Component {
     state = {
-        isOpen: this.props.isOpen
+        isOpen: (this.props.isOpen === undefined) ? false : this.props.isOpen
```
indeed, it is.

bastilian commented on src/SmartComponents/NotificationEdit/NotificationEdit.js:
```diff
@@ -87,7 +86,8 @@ NotificationEdit.propTypes = {
     fetchEndpoint: PropTypes.func.isRequired,
-    match: PropTypes.object
+    match: PropTypes.object,
+    store: PropTypes.object
```
another leftover from trying to get tests work.

adamruzicka commented on src/SmartComponents/NotificationsIndex/NotificationIndex.test.js:
```diff
@@ -0,0 +1,54 @@
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('it takes endpoints', () => {
```
Too many `it`s

bastilian replied:
Removed `it`.

adamruzicka commented:
Thanks @bastilian !

josejulio committed:
WIP

josejulio committed:
Update with latest backend changes

josejulio committed:
Connects enable/disable endpoint (integration)

josejulio committed:
Connect with delete endpoint api call

josejulio committed:
Using edit api call

josejulio committed:
- New integrations are enabled
- Fixed the use of the correct endpoints for enable/disable

josejulio committed:
Refactor DeleteModal into components

josejulio committed:
Add selectos to use for ouiacomponents

josejulio committed:
- Removing ouia from DeleteModal - inner component were using portal and
 thus not easy to follow, the ouia was meaningless in this case.
- Adding more tests

josejulio committed:
Refactor DeleteModal into an ActionModal to be able to use it for saving

josejulio committed:
Refactor CreatePage to use the ActionModal/SaveModal

josejulio committed:
IntegrationAdapter tests

josejulio committed:
Tests on *Modal

josejulio committed:
Basename tests

josejulio committed:
IntegrationModal* tests

josejulio committed:
Adapt OuiaSelectors to closely match the testing-library query functions
 - missing the promise based functions (findBy and findAllBy)

josejulio committed:
Moving common code to insights-common-typescript

josejulio committed:
Adding exports formats for Integrations

josejulio committed:
Remove types for now, we only have webhook

josejulio committed:
Minor changes in text

josejulio committed:
Adds more data to Integration (from webhook)

josejulio committed:
Fix lint and test errors

josejulio committed:
Adding additional webhook fields to the form

josejulio committed:
Adding the failure/degraded inline alerts on the table

josejulio committed:
Adding "Last connection attempt" to table

josejulio committed:
Adding tests to IntegrationsTable

josejulio committed:
Updating colors and fixing a bug where the Integration could be a string due to a call to the server

theute commented:
Can we merge this ?

josejulio commented:
> Can we merge this ?

Yes. Merging now that we have a backend in place.
